### PR TITLE
mudlet: 3.0.0-delta -> 4.0.3 (fixes broken package)

### PR DESCRIPTION
**Pull request by pstn**
###### Motivation for this change
mudlet has been broken for a while now. I managed to fix and upgrade it to the most recent version. 

I added myself as a maintainer since nobody else seems to have interest in that package any more.

Thanks to @abbradar for the help and the patch to upstream.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @wyvie

### CODE REVIEW

**Comment by pstn** [comment]
The current maintainer has been inactive for quite some time, so please consider reviewing this even if you are not them.

File: pkgs/games/mudlet/default.nix
Comment (teto) [inline pkgs/games/mudlet/default.nix]: have you tried using an environment  instead? `luaEnv = lua.withPackages(p: [luaFileSystemPath ...])`. Will make the path simpler and indenpendant of the lua version, like one could override with lua52 and still have the paths valid instead of `luaUtf8Path = "${luautf8}/lib/lua/5.1/?.so";`

**Comment by pstn** [comment]
Removed a few dependencies that became unneeded but I forgot to clean up.

File: pkgs/games/mudlet/default.nix
Comment (pstn) [reply pkgs/games/mudlet/default.nix]: I didn't because I didn't know about that method. You are right. It's way cleaner.
I couldn't get rid of all dependencies on the lua version, but I moved that version in a variable at the top for easier maintenance and discovery.

File: pkgs/games/mudlet/default.nix
Comment (abbradar) [inline pkgs/games/mudlet/default.nix]: Perhaps this is not needed now that we use `withPackages`?

File: pkgs/games/mudlet/default.nix
Comment (abbradar) [inline pkgs/games/mudlet/default.nix]: Use `"NO"` here too for consistency and because I'm not sure `false` would work but `"NO"` is used by AUR package.

File: pkgs/games/mudlet/default.nix
Comment (pstn) [reply pkgs/games/mudlet/default.nix]: `preConfigure` removed.

**Review by abbradar — APPROVED** [review]
@teto, can you check Lua stuff once more just in case before we merge?

File: pkgs/games/mudlet/default.nix
Comment (pstn) [reply pkgs/games/mudlet/default.nix]: Done.

File: pkgs/games/mudlet/default.nix
Comment (pstn) [reply pkgs/games/mudlet/default.nix]: Scratch that. Got it working by overwriting the lua version in `all-packages.nix`.

File: pkgs/games/mudlet/default.nix
Comment (teto) [inline pkgs/games/mudlet/default.nix]: may be best for the patch to live in nixpkgs ?

**Review by teto — CHANGES_REQUESTED** [review]
there are too many changes in the lua generated packages list. I wonder if some of them are linked with the luarocks update.
Many seem to have changed to luafr.org, which is a mirror of the luarocks server. Are you based in france by any chance @pstn ? This is something I need to fix in luarocks2nix Meanwhile you could just restrict the changes in in generated-packages.nix to only the changes you need (with git add -p).

File: pkgs/games/mudlet/default.nix
Comment (abbradar) [reply pkgs/games/mudlet/default.nix]: They already merged it upstream actually: https://github.com/Mudlet/Mudlet/commit/6306f99a847416dc8a690e57552553136ca77ea6

**Comment by pstn** [comment]
No, I'm in Germany.

I did discard all the unneeded changes in both commits. Thanks for your help.

File: pkgs/games/mudlet/default.nix
Comment (pstn) [reply pkgs/games/mudlet/default.nix]: I had it in nixpkgs but @abbradar wanted me to pull it from their repo.

I think i like it better this way because you could access more information about the patch when you know how to handle the link.

Comment (teto) [reply pkgs/games/mudlet/default.nix]: the issue is that url points towards @abbradar url. so if he deletes his fork or the branch, the package will stop compiling. If the patch were in nixpkgs then we are in control. Referencing the upstreaming patch should be fine as if their commit disappear, it means we will have problems anyway.

**Comment by teto** [comment]
No, it's me thanking you ! I wanted this package to use lua environments for at least a year but fixing it proved too difficult. Not only did you do it but you also found some issue in how the lua packages are generated. I have a fix I will submit soon.
The last thing I would like to see before merging is to fetch the patch from the mudlet's repo. Thanks for keeping up with us.

File: pkgs/games/mudlet/default.nix
Comment (pstn) [reply pkgs/games/mudlet/default.nix]: Ah, got it. fetching the patch from mudlet/mudlet now.

**Comment by teto** [comment]
@GrahamcOfBorg build luaPackages.mudlet

File: pkgs/games/mudlet/default.nix
Comment (pstn) [reply pkgs/games/mudlet/default.nix]: Ah, got it. fetching the patch from mudlet/mudlet now.

**Comment by teto** [comment]
not sure why it's stuck, I ran a nix-review on x86-64 and it worked. Merging. Thanks to all those involved.

**Pull request merged**
